homeassistant/components/aurora/config_flow.py

```diff
@@ -0,0 +1,110 @@
+"""Config flow for SpaceX Launches and Starman."""
+import logging
+
+from auroranoaa import AuroraForecast
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_THRESHOLD, DEFAULT_NAME, DEFAULT_THRESHOLD, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for NOAA Aurora Integration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            name = user_input[CONF_NAME]
+            longitude = user_input[CONF_LONGITUDE]
+            latitude = user_input[CONF_LATITUDE]
+
+            session = aiohttp_client.async_get_clientsession(self.hass)
+            api = AuroraForecast(session=session)
+
+            try:
+                await api.get_forecast_data(longitude, latitude)
+            except ConnectionError:
```

Comment:
The aurora library is using aiohttp. `ConnectionError` is not an aiohttp exception. We should probably either catch an aiohttp error or a library specific error here.

https://docs.aiohttp.org/en/stable/client_reference.html#client-exceptions

Reply:
Addressed in PR https://github.com/home-assistant/core/pull/43148